### PR TITLE
Exercise 4 iterator, promise, prototype

### PR DESCRIPTION
**Pull request by AkashDPA**

### CODE REVIEW

File: Week-4/iterator.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        let i = 1;
+        let thisNum = 0, nextNum = 0;
+        return {
+            next: () => {
+                i++;
+                if(i > n){
+                    return { done: true }
+                }
+                thisNum = nextNum;
+                nextNum = (thisNum + nextNum) || 1;
```
Comment (manishsundriyal) [inline Week-4/iterator.js:19]:
```suggestion
          [thisNum, nextNum] = [nextNum, (thisNum + nextNum) || 1];
```

**Review by manishsundriyal — CHANGES_REQUESTED** [review]
Please look at the suggested changes

File: Week-4/iterator.js (hunk 1 above)
Comment (AkashDPA) [reply Week-4/iterator.js]: fixed in [feb55eb](https://github.com/pesto-students/p8-manish-AkashDPA/pull/5/commits/feb55eb5318a7bcaa785ab2fe5448998c704c6da)

**Review by manishsundriyal — APPROVED** [review]
LGTM

**Pull request merged**